### PR TITLE
Second Batch of Submissions Tests

### PR DESCRIPTION
**Pull request by 20wildmanj**
## Description

- Fix some variable naming and remove misleading comment in submissions_controller#missing
- add `create_course_no_submissions_hash` to have a test course with no submissions
- add the tests in the second batch of #1803

## How Has This Been Tested?

Run the tests, see that they pass

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have run rubocop and erblint for style check. If you haven't, run `overcommit --install && overcommit --sign` to use pre-commit hook for linting
- [ ] My change requires a change to the documentation, which is located at [Autolab Docs](https://docs.autolabproject.com/)
- [ ] I have updated the documentation accordingly, included in this PR

### CODE REVIEW

File: spec/controllers/submissions_controller_spec.rb [hunk 1]
```diff
@@ -209,4 +255,151 @@
       it_behaves_like "edit_failure"
     end
   end
+
+  describe "#downloadAll" do
+    include_context "controllers shared context"
+    context "when user is Instructor of class with submissions" do
+      let!(:user) { instructor_user }
+      it "downloads all submissions for an assessmetn" do
```
Comment (jlge) [inline spec/controllers/submissions_controller_spec.rb:263]: nit: typo

File: spec/controllers/submissions_controller_spec.rb [hunk 2]
```diff
@@ -209,4 +255,151 @@
...
+        end
+      end
+    end
+  end
+
+  describe "#missing" do
+    include_context "controllers shared context"
+    context "when user is student" do
+      let!(:user) { student_user }
+      it "gets missing submissions" do
```
Comment (jlge) [inline spec/controllers/submissions_controller_spec.rb:318]: maybe I'm understanding wrong but should this be doesn't get missing submissions bc it's expecting the response to fail

File: spec/controllers/submissions_controller_spec.rb [hunk 3]
```diff
@@ -209,4 +255,151 @@
...
+        sign_in(user)
+        get :missing, params: { course_name: @course.name, assessment_name: @assessment.name }
+        expect(response).not_to be_successful
+        expect(response.body).not_to match(/Missing Submissions/m)
+      end
+    end
+    context "when user is Instructor of class with submissions" do
+      let!(:user) { instructor_user }
+      it "fails to render" do
```
Comment (jlge) [inline spec/controllers/submissions_controller_spec.rb:327]: nit: think this test name can be a little more descriptive

**Review by jlge — APPROVED** [review]
lgtm! left some nits about test naming

File: spec/controllers/submissions_controller_spec.rb (hunk 2 above)
Comment (20wildmanj) [reply spec/controllers/submissions_controller_spec.rb:318]: Indeed, the name is inaccurate.

**Pull request merged**